gradoop-flink/src/test/java/org/gradoop/flink/io/impl/csv/CSVDataSinkTest.java

```diff
@@ -69,6 +69,26 @@ public void testWriteWithDifferentPropertyTypes() throws Exception {
     checkCSVWrite(tmpPath, loader.getLogicalGraphByVariable("edges"));
   }
 
+  /**
+   * Test CSVDataSink to properly separate the metadata
+   * of edges and vertices using the same label.
+   *
+   * @throws Exception
+   */
+  @Test
+  public void testWriteWithSameLabel() throws Exception {
+    String tmpPath = temporaryFolder.getRoot().getPath();
+
+    // The properties are incompatible to get a conversion error
+    // if the metadata is not separated
+    FlinkAsciiGraphLoader loader = getLoaderFromString(
+      "graph[" +
```

Comment:
Pls add multiple entities with same label to test that case as well. 
Your example Graph posted within your Issue #882 would fit.  
```
test[
(v1:A {keya:1, keyb:2, keyc:\"Foo\"}),
(v2:A {keya:1.2f, keyb:\"Bar\", keyc:2.3f}),
(v3:A {keya:\"Bar\", keyb:true}),
(v1)-[e1:A {keya:14, keyb:3, keyc:\"Foo\"}]->(v1),
(v1)-[e2:a {keya:1.1f, keyb:\"Bar\", keyc:2.5f}]->(v1),
(v1)-[e3:a {keya:true, keyb:3.13f}]->(v1)
]
```
- Multiple vertex entities with label 'A'
- Multiple edge entities with label 'a'

Reply:
The problem with the `test` graph is it doesn't fail reliably. It only sometimes fails because edge e1 is compatible to the metadata of v:A but not the other way around. The 'e:a' edges don't have any 'v:a' vertices to collide with, only edge 'e1:A' conflicts with the vertices.
```
graph[
(v1:A {keya:true, keyb:1, keyc:\"Foo\"}),
(v2:A {keya:false, keyb:2.1f}),
(v3:B {keya:2.3f, keyb:\"Bar\"}),
(v4:B {keya:1.1d}),
(v1)-[e1:A {keya:1, keyb:\"Foo\", keyc:3.3d}]->(v2),
(v2)-[e2:A {keya:2, keyb:7.2d}]->(v1),
(v2)-[e3:B {keya:false}]->(v3),
(v3)-[e4:B {keya:true, keyb:13}]->(v4)
]
```
This should be guaranteed to fail on 'keya' with a conversion error and adds multiple entities.

Reply:
Allright, then use your proposed graph for testing.